---
title: Analysis of GC-biased Gene Conversion Using Haplotype Networks
excerpt: "A research project done during the Bioinformatics Internship and Training Program of the Philippine Genome Center (PGC)."
text1: "[Summary]"
text2: "[Github Repository]"
text3: "[Narrative Report]"
link1: "https://edwardnataniel.github.io/blog/gBGC/"
link2: "https://github.com/edwardnataniel/bioinformatics-gbgc-project"
link3: "https://edwardnataniel.github.io/files/gBGC_Narrative_Report.pdf"
modified: 04/6/2020, 15:00:00
tags: [intro, beginner, jekyll, tutorial]
comments: true
category: blog
---
#### Full narrative report can be accessed here:
<a href="https://edwardnataniel.github.io/files/gBGC_Narrative_Report.pdf"> https://edwardnataniel.github.io/files/gBGC_Narrative_Report.pdf </a>


## Overview

**GC-biased gene conversion (gBGC)** is a recombination-associated process where **G/C alleles are favored over A/T alleles**. The non-symmetric segregation of G/C and A/T is a result of the mismatch repair bias that occurs during **meiosis**, producing more G/C than A/T-containing gametes. This mimics positive selection by promoting the fixation of G/C over A/T, thereby having implications on studies on functional genomics and evolution. However, despite the existence of evidence indicating the presence of gBGC in mammals, and its effect on evolution, **there is still no direct classical statistical test of gBGC on humans**. This paper aims **to test the hypothesis of GC-biased gene conversion in the highly recombining regions of human DNA**. Candidate hotspot regions were determined and CpG islands, gene, and gene-associated regions were eliminated. **Haplotype networks** were constructed for each of these regions; and recombination and gene conversion events were inferred from these networks. Statistical tests were performed on the inferred gene conversion mutations to test for the gBGC hypothesis. Results show that **there is a significant (p<0.05)  bias towards AT->GC mutations**.

<img src="https://edwardnataniel.github.io/images/projects/5/meiosis.PNG" width="250" />
  
Fig.1 Meiosis

<img src="https://edwardnataniel.github.io/images/projects/5/geneconversion.PNG" width="500" />
Fig.2 Gene Conversion

## The gBGC Hypothesis
Gene conversion events are biased towards GC Alleles. That is,
#### AT ‚Üí GC  >  GC ‚Üí AT
(Presence of statistical evidence that AT to GC conversions are greater than GC to AT conversions.)

The **primary objective** of this study is to test the gBGC hypothesis. 

## Methodology
1. **Acquire and analyze a genetic map** for the human genome and identify regions with high recombination rates (recombination hotspots).
2. **Filter data** to remove genes, CpG islands, and other confounding factors.
3. **Download sequence data** from the 1000 Genomes Project corresponding to these regions, with (low-recombining) flanking regions as a control.
4. **Construct haplotype networks** based on the sequence data.
5. **Infer gene conversion events**, and collect data on associated mutations.
6. **Perform statistical testing** of stated hypothesis

### Step 1: Acquire and analyze a genetic map for the human genome and identify regions with high recombination rates

* Recombination data (GRCh38 / build38) were downloaded from the **NCBI website**.

* A **Python script** was written to compute the recombination rates (in cM/MB) and to identify regions with high recombination rates (rate of at least 15 cM/MB and length of at least 3000 bp).

* The left and right flanking regions were also selected ( with maximum distance of 3000bp from the candidate hotspot region).

<img src="https://edwardnataniel.github.io/images/projects/5/step1.PNG" width="400" />

### Step 2: Filter data to remove genes, CpG islands, and other confounding factors.

These regions may confound the analysis so they need to be filtered out. 

* **Genes and CpG islands data** were downloaded from the UCSC Genome Browser.
* A **Python script** was written to convert the raw data in **BED** (tabular) format.
* Bedtools, a toolset for genome arithmetic, was run to subtract regions of genes and CpG islands from the hotspot region.

<img src="https://edwardnataniel.github.io/images/projects/5/step2.PNG" width="400" />

### Step 3: Download sequence data from the 1000 Genomes Project using Tabix.

* A **wrapper script** was written to download data (in **VCF format**) from the IGSR: The international Genome Sample Resource using **Tabix**.

### Step 4: Construct haplotype networks based on the sequence data.

* A code written in **R** was written to construct:
  - **DNAbin** from VCF files
  - **Haplotypes** from DNAbins
  - **Haplotype Networks** from Haplotypes

<img src="https://edwardnataniel.github.io/images/projects/5/step4.PNG" width="400" />

### Step 5: Infer gene conversion events, and collect data on associated mutations.

* The mutations that resulted from the inferred recombination and gene conversion events were tallied.

### Step 6: Perform statistical testing of stated hypothesis.

* A Chi-square test for goodness-of-fit was performed to test for the presence of GC bias, where the null hypothesis was 1:1 ratio of AT->GC and GC->AT. 

<img src="https://edwardnataniel.github.io/images/projects/5/step6.PNG" width="200" />

### Master script
* A master script that will run all the steps above was written to automate the whole process. This code runs in the **high performance computing (hpc)** server of the Philippine Genome Center. 

 
<img src="https://edwardnataniel.github.io/images/projects/5/master.PNG" width="600" />

## Results
* A total of 340 hotspots, 319 left flanking regions, and 315 right flanking regions were successfully sampled. 
* We also counted the conversion event of each of the regions (hotspots = 1168, left flanks = 1475, and right flanks = 1149). The number of mutations was also tallied.
* The results show that **the number of  AT -> GC mutations in all regions is significantly larger than the number of GC -> AT mutations**, with a total of 19,546 AT -> GC mutations and 17,614 GC -> AT mutations. 

<img src="https://edwardnataniel.github.io/images/projects/5/results.PNG" width="450" />

### Statistical Test 
* The result shows that there is sufficient evidence to say, at 0.05 level of significance, that **there is bias in gene conversion in favor of AT -> GC than GC -> AT in across all regions**  (ùúí2 = 100.45, degrees of freedom = 1, p-value = 2.2e-16). 
* We also performed a Chi-square goodness-of-fit test for each of the region type - hotspots, left flank and right flank. **All tests reject the null hypothesis**.

## Discussion
* Previous studies (Fullerton et. al., 2001;   Arbeithuber et. al., 2015) suggest that gene conversion events are more likely to occur in highly recombining regions. However, our results show that **the location of gene conversion does not discriminate relative to value of recombination rate**. GC-biased gene conversion was also observed in areas with relatively low recombination rates. It must be noted, however, that **several factors may have brought about this observation**. For example, correlation between sample length and number of loci must be checked. Given that the number of subnetworks calculated is dependent on the number of variants observed, it may be that the number of inferred events.

* Despite the limitations in this study, it provides a preliminary data that **shows a strong and significant bias towards GC alleles over AT alleles**.
